src/nomad_simulation_parsers/parsers/lammps/file_parsers.py

```diff
@@ -0,0 +1,771 @@
+                with open(file_path, 'rb') as file:
+                    file_header = file.read(header_size)
+                    file_header_str = file_header.decode(errors='ignore')
+            except Exception:
+                file_header_str = ''
+            return re.search(regex_pattern, file_header_str)
+
+        read_data = self.get('read_data')
+        # TODO: Check in with Leo on the reasoning behind this change
+        # # Chop out 'CPU' before, then just check none
+        # if read_data is not None:
+        #     try:
+        #         read_data.remove('CPU')
+        #     except Exception:
+        #         pass
+        if read_data is None or 'CPU' in read_data:
+            self.logger.warning('Data file not specified in directory, will scan.')
+            for ext in self.DATA_FILE_PATTERNS:
+                data_files = search_files(
+                    pattern=f'*{ext}*',
+                    basedir=self.maindir,
+                    deep=False,  # Only search current directory
+                )
+            if not data_files:
+                # Search any file for the LAMMPS data file header.
+                # Fallback to the LAMMPS input structure, if no run data file is found.
+                data_files = os.listdir(self.maindir)
+                patterns = ['LAMMPS data file', 'LAMMPS Description']
+                for pattern in patterns:
+                    data_files = [
+                        f for f in data_files if check_file_header(f, pattern)
+                    ]
+                    if data_files:
+                        break
+            if len(data_files) > 1:
+                # Search data files for the LAMMPS data file header.
+                data_files = [
+                    f for f in data_files if check_file_header(f, 'LAMMPS data file')
+                ]
+        else:
+            data_files = read_data
+
+        if not data_files:
+            self.logger.warning('No data_files found to match the log file.')
+
+        return [os.path.join(self.maindir, f) for f in data_files]
+
+    def get_pbc(self) -> list[str]:
+        pbc = self.get('boundary', ['p', 'p', 'p'])
+        return [v == 'p' for v in pbc]
+
+    def get_sampling_method(self) -> tuple[str, str]:
+        fix_style = self.get('fix', [[''] * 3])[0][2]
+
+        sampling_method = (
+            'langevin_dynamics' if 'langevin' in fix_style else 'molecular_dynamics'
+        )
+        return sampling_method, fix_style
+
+    def get_thermostat_settings(self) -> dict:
+        fix = self.get('fix', [[]])[0]
+
+        try:
+            fix_style = fix[2]
+        except IndexError:
+            return {}
+
+        temp_unit = self.units.get('temperature', 1)
+        press_unit = self.units.get('pressure', 1)
+        time_unit = self.units.get('time', 1)
+
+        res = dict()
+        if fix_style.lower() == 'nvt':
+            try:
+                res['target_T'] = float(fix[5]) * temp_unit
```

Comment:
if possible, convert fix to list of floats so we do not see these float() conversions